machine-runner/src/runner/runner.ts

```diff
@@ -647,21 +661,72 @@ namespace MachineRunnerIterableIterator {
     MachineName extends string,
     StateUnion extends unknown,
   >({
+    emitter,
     nextValueAwaiter,
     destruction,
+    internals,
   }: {
+    emitter: MachineEmitter<SwarmProtocolName, MachineName, StateUnion>
+    internals: RunnerInternals.Any
     nextValueAwaiter: NextValueAwaiter<
       StateOpaque<SwarmProtocolName, MachineName, string, StateUnion>
     >
     destruction: Destruction
   }): MachineRunnerIterableIterator<SwarmProtocolName, MachineName, StateUnion> => {
-    const onThrowOrReturn = async (): Promise<
-      IteratorResult<StateOpaque<SwarmProtocolName, MachineName, string, StateUnion>, null>
-    > => {
+    type SO = StateOpaque<SwarmProtocolName, MachineName, string, StateUnion>
+
+    const internalsCurrentIsOkForActual = () =>
+      CommandGeneratorCriteria.allOk({
+        isCaughtUp: () => internals.caughtUp && internals.caughtUpFirstTime,
+        isQueueEmpty: () => internals.queue.length === 0,
+        isNotExpired: () => !ImplStateOpaque.isExpired(internals, internals.current),
+        isNotLocked: () => !ImplStateOpaque.isCommandLocked(internals),
+        isNotDestroyed: () => !ImplStateOpaque.isRunnerDestroyed(internals),
+      })
+
+    const onThrowOrReturn = async (): Promise<IteratorResult<SO, null>> => {
       destruction.destroy()
       return Promise.resolve({ done: true, value: null })
     }
 
+    const actual = (): Promise<IteratorResult<SO, null>> => {
+      if (destruction.isDestroyed()) {
+        return Promise.resolve({ done: true, value: null })
+      }
+
+      if (internalsCurrentIsOkForActual()) {
+        return Promise.resolve({
+          done: false,
+          value: ImplStateOpaque.make(internals, internals.current),
+        })
+      }
+
+      const peekOrUnlockPromise = ((): Promise<IteratorResult<SO, null>> => {
```

Comment:
this level of nesting doesn’t buy anything

Reply:
It buys a focusing scope saying that these group of code relates tighter than what's outside of it. It also makes it clearer that `actual` has 3 main paths:
- when the iterator is destroyed
- when state is caught up
- when state isn't caught up

Reply:
Removed the nesting because `peek` is not used anymore.
Reason for not using peek is that it has a state inside NextValueAwaiter.
The state keeps track whether the machine's stateopaque is consumed or not.
The state isn't affected by a call to `actual`, nor by the command locking mechanism.

A confusion happens inside actual function where when a command is locked, `peek` still can return the stored state from NextValueAwaiter.